cmd/cstor-volume-mgmt/controller/volume-controller/after_queue_handler.go

```diff
@@ -0,0 +1,159 @@
+package volumecontroller
+
+import (
+	"fmt"
+	"os"
+	"reflect"
+
+	"github.com/golang/glog"
+	"github.com/openebs/maya/cmd/cstor-volume-mgmt/controller/common"
+	"github.com/openebs/maya/cmd/cstor-volume-mgmt/volume"
+	apis "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	"github.com/openebs/maya/pkg/util"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/tools/cache"
+)
+
+// syncHandler compares the actual state with the desired, and attempts to
+// converge the two. It then updates the Status block of the cStorVolumeUpdated resource
+// with the current status of the resource.
+func (c *CStorVolumeController) syncHandler(key string, operation common.QueueOperation) error {
+	glog.Infof("at sync handler")
+	cStorVolumeGot, err := c.getVolumeResource(key)
+	if err != nil {
+		return err
+	}
+	status, err := c.cStorVolumeEventHandler(operation, cStorVolumeGot)
+	if status == common.CVStatusIgnore {
+		return nil
+	}
+	cStorVolumeGot.Status.Phase = string(status)
+	if err != nil {
+		_, err := c.clientset.OpenebsV1alpha1().CStorVolumes().Update(cStorVolumeGot)
+		if err != nil {
+			return err
+		}
+		return err
+	}
+	_, err = c.clientset.OpenebsV1alpha1().CStorVolumes().Update(cStorVolumeGot)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// cStorVolumeEventHandler is to handle cstor volume related events.
+func (c *CStorVolumeController) cStorVolumeEventHandler(operation common.QueueOperation, cStorVolumeGot *apis.CStorVolume) (common.CStorVolumeStatus, error) {
+	volume.FileOperatorVar = util.RealFileOperator{}
+	volume.UnixSockVar = util.RealUnixSock{}
+	glog.Infof("%v event received for volume : %v ", operation, cStorVolumeGot.Spec.VolumeName)
+	switch operation {
+	case common.QOpAdd:
+		// CheckValidVolume is to check if volume attributes are correct.
+		err := volume.CheckValidVolume(cStorVolumeGot)
+		if err != nil {
+			return common.CVStatusOffline, err
+		}
+
+		err = volume.CreateVolume(cStorVolumeGot)
+		if err != nil {
+			return common.CVStatusFailed, err
+		}
+		break
```

Comment:
break not required

Reply:
I don't know if it is not required. I feel safe if it is there. Is that OK ?

Reply:
Sure. Go switches are `break`ed by default.